### ISSUE TITLE
cannot resolve symbol 'RAND_bytes'

### ISSUE THREAD

**Issue by gtuxyco** [bungle/lua-resty-session#31]
Hi! i'm try to use this library with the Nginx Openresty For Windows and show me this error.

> 
2016/08/30 15:36:29 [error] 9084#3740: *1 lua entry thread aborted: runtime error: C:\Users\pablo\Desktop\nginx-1.11.3//resty\session.lua:30: cannot resolve symbol 'RAND_bytes': No se encontr󠥬 proceso especificado.

**Comment by bungle** [comment]
@gtuxyco, Do you have OpenSSL build with it?

**Comment by gtuxyco** [comment]
here the output.
nginx version: nginx/1.11.3

I use this build of nginx 
https://github.com/LomoX-Offical/nginx-openresty-windows

Thanks for the help :+1:

**Comment by bungle** [comment]
I meant output with capital V -> `nginx -V`, not `nginx -v`.

**Comment by gtuxyco** [comment]
nginx version: nginx/1.11.3
built by cl
built with OpenSSL 1.0.2h  3 May 2016
TLS SNI support enabled
configure arguments: --with-cc=cl --builddir=objs --prefix= --conf-path=conf/nginx.conf --pid-path=logs/nginx.pid --http-log-path=logs/access.log --error-log-path=logs/error.log --sbin-path=nginx.exe --http-client-body-temp-path=temp/client_body_temp --http-proxy-temp-path=temp/proxy_temp --http-fastcgi-temp-path=temp/fastcgi_temp --http-scgi-temp-path=temp/scgi_temp --http-uwsgi-temp-path=temp/uwsgi_temp --with-cc-opt=-DFD_SETSIZE=32768 --with-pcre=objs/lib/pcre-8.38 --with-zlib=objs/lib/zlib-1.28 --with-openssl=objs/lib/openssl-1.0.2h --with-select_module --with-http_realip_module --with-http_addition_module --with-http_sub_module --with-http_dav_module --with-http_stub_status_module --with-http_flv_module --with-http_mp4_module --with-http_gunzip_module --with-http_gzip_static_module --with-http_auth_request_module --with-http_random_index_module --with-http_secure_link_module --with-http_slice_module --with-mail --with-stream --with-http_ssl_module --with-mail_ssl_module --with-stream_ssl_module --with-http_v2_module --with-ipv6 --add-module=../ngx_devel_kit-0.3.0 --add-module=../echo-nginx-module-0.59 --add-module=../ngx_coolkit-0.2rc3 --add-module=../set-misc-nginx-module-0.30 --add-module=../ngx_postgres-1.0rc7 --add-module=../form-input-nginx-module-0.12 --add-module=../encrypted-session-nginx-module-0.05 --add-module=../ngx_lua_module-0.10.5 --add-module=../ngx_lua_upstream-0.05 --add-module=../headers-more-nginx-module-0.30 --add-module=../array-var-nginx-module-0.05 --add-module=../nginx-http-concat-module --add-module=../rds-json-nginx-module-0.14 --add-module=../redis2-nginx-module-0.13

**Comment by bungle** [comment]
@gtuxyco,

Ok thanks. Do you have same problem with `lua-resty-string`'s [random](https://github.com/openresty/lua-resty-string/blob/master/lib/resty/random.lua)?

e.g.

```nginx
content_by_lua_block {
    local r = require "resty.random"
    ngx.say(r.bytes(10, true))
}
```

**Comment by bungle** [comment]
@gtuxyco,

I also did a few modifications, can you replace your `session.lua` with this version:
https://github.com/bungle/lua-resty-session/blob/master/lib/resty/session.lua

**Comment by gtuxyco** [comment]
Hi!, i try the resty.random call and work fine, this generate a random text 

> /qmrf5set7PYOA==
replace session.lua but remained the same error 

> 2016/08/31 23:58:30 [error] 2992#4844: *1 lua entry thread aborted: runtime error: C:\Users\pablo\Desktop\nginx-1.11.3//resty\session.lua:29: cannot resolve symbol 'RAND_bytes': No se encontró el proceso especificado.

**Comment by bungle** [comment]
Huh, this is really weird. We are close to the solution, but not quite there yet.

**Comment by bungle** [comment]
@gtuxyco,

I removed my FFI stuff, because I think it is faster to solve this issue by using resty.random, and it is right thing to do anyway. So I just committed a new version, can you try this out:
https://github.com/bungle/lua-resty-session/blob/master/lib/resty/session.lua


P.S: I don't understand why it didn't work on your machine, but resty.random did. They are practically same code. But well, please try the new one, and report back if it resolves this issue. I will then make a new release.

**Comment by gtuxyco** [comment]
@bungle Solved with the last patch. I don't know either but now is seen to work excellent :+1:

**Comment by bungle** [comment]
@gtuxyco, ok, good to know, I will release a new version then.

**Comment by bungle** [comment]
@gtuxyco, I have now released 2.9. Thanks again for reporting this. I will close this issue now, as I consider it resolved. Please reopen if you have any further problems.
